src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
+  Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+  bite(animal) {
+    if (!animal.hidden && animal instanceof Herbivore) {
+      animal.health = animal.health - 50;
+    }
+
+    if (animal.health <= 0) {
```

Comment:
The check for animal health and removal of dead animals from the alive array should not be inside the `bite` method. It should be in the Animal class and get called whenever an animal's health changes.